### ISSUE TITLE
lstm output shape problem when reshaping lstm input

### ISSUE THREAD

**Issue by Jackal93** [keras-team/keras#9419]
I'm having trouble understanding what's going on in the following code, in which i'm reshaping the input from `(batch_size, features)` to `(1, batch_size, features)`, then feeding this to an LSTM with <code>return_sequences=True</code> (i.e. i'm basically turning the batch into time steps).

I'm under the impression that this should output something with shape <code>(1, time_steps, lstm_units)</code>, where <code>batch_size == time_steps</code>, and indeed Keras' model summary confirms this, however the actual output has shape <code>(time_steps, time_steps, lstm_units)</code>.

Moreover the output is identical across the first dimension.

    

    import numpy as np
    import keras.backend as K
    from keras.layers import Input, LSTM, Lambda
    from keras.models import Model
    
    
    batch_size = 2
    
    batch_shape = (batch_size, 5)
    inp = Input(batch_shape=batch_shape)
    re = Lambda(lambda x: K.reshape(x, (1, batch_size, 5)))
    lstm = LSTM(1, stateful=True, return_sequences=True)
    out = lstm(re(inp))
    
    model = Model(inputs=inp, outputs=out)
    
    model.summary()
    
    x = np.random.random(batch_shape)
    
    y = model.predict(x, batch_size=batch_size)
    print("y.shape:", y.shape)
    print('y:', y, sep='\n')

This outputs:

    _________________________________________________________________
    Layer (type)                 Output Shape              Param #   
    =================================================================
    input_1 (InputLayer)         (2, 5)                    0         
    _________________________________________________________________
    lambda_1 (Lambda)            (1, 2, 5)                 0         
    _________________________________________________________________
    lstm_1 (LSTM)                (1, 2, 1)                 28        
    =================================================================
    Total params: 28
    Trainable params: 28
    Non-trainable params: 0
    _________________________________________________________________
    
    y.shape: (2, 2, 1)
    y:
    [[[-0.11784139]
      [-0.16627716]]
    
     [[-0.11784139]
      [-0.16627716]]]

As i said the output is identical across the first dimension, i.e. -0.11784139 and -0.16627716 are repeated, which seems suspicious as the input is random.

Why is this happening? Why is <code>y.shape</code> not <code>(1,2,1)</code>?

**Comment by Jackal93** [comment]
I don't understand. I tried removing the `batch_size` in the input shape but the result is the same.
You say the `batch_size` is assumed not to be change, does that mean my reshape is the problem?

**Comment by rex-yue-wu** [comment]
Yes, I think your reshape is problematic. What prevent you to directly use a 3D input? I mean why not skip the `reshape` layer (reshaping a 2D input as 3D), while directly preparing your input in 3D.

**Comment by Jackal93** [comment]
Nothing is preventing me from using directly 3D input, but why is that reshape problematic? There is a conflict between the output shape communicated by Keras and the actual output shape you get based on the input shape, which I don't think there should be.

**Comment by tRosenflanz** [comment]
Your axis of reshape is wrong cahnge 
`re = Lambda(lambda x: K.reshape(x, (1, batch_size, 5))) `
to
` re = Lambda(lambda x: K.reshape(x, (batch_size, 1, 5)))`

Keras assume that shapes have format of (batch_size, num_timesteps, num_features/timestep)  so LSTM operates on axis 1 and not 0 

Also in future try to post this questions on StackOverflow since this is not an implementation issue and not a bug of Keras

**Comment by Jackal93** [comment]
Ok, thanks.
